src/components/course/CourseAssociatedPrograms.jsx

```diff
@@ -38,23 +36,40 @@ const CourseAssociatedPrograms = () => {
               </div>
             </div>
             <div className="col">
-              <Hyperlink
-                destination={features.ENABLE_PROGRAMS ? `/${enterpriseCustomer.slug}/program/${program.uuid}`
-                  : program.marketingUrl}
-                target="_blank"
-                onClick={() => {
-                  sendEnterpriseTrackEvent(
-                    enterpriseCustomer.uuid,
-                    'edx.ui.enterprise.learner_portal.course.sidebar.program.clicked',
-                    {
-                      program_title: program.title,
-                      program_type: program.type,
-                    },
-                  );
-                }}
-              >
-                {program.title}
-              </Hyperlink>
+              {features.ENABLE_PROGRAMS ? (
+                <Link
+                  to={`/${enterpriseCustomer.slug}/program/${program.uuid}`}
+                  onClick={() => {
+                    sendEnterpriseTrackEvent(
+                      enterpriseCustomer.uuid,
+                      'edx.ui.enterprise.learner_portal.course.sidebar.program.clicked',
+                      {
+                        program_title: program.title,
+                        program_type: program.type,
+                      },
+                    );
+                  }}
+                >
+                  {program.title}
+                </Link>
+              ) : (
+                <Hyperlink
+                  destination={program.marketingUrl}
+                  target="_blank"
+                  onClick={() => {
+                    sendEnterpriseTrackEvent(
+                      enterpriseCustomer.uuid,
+                      'edx.ui.enterprise.learner_portal.course.sidebar.program.clicked',
+                      {
+                        program_title: program.title,
+                        program_type: program.type,
+                      },
+                    );
+                  }}
+                >
+                  {program.title}
+                </Hyperlink>
```

Comment:
So, in this case, `program.marketingUrl` is to an external website, not an internal Learner Portal route. However, because `features.ENABLE_PROGRAMS` is always true for us in practice, the `Link` will actually be used in practice. This keeps the previous implementation backwards compatible to support the fallback to the `program.marketingUrl` (i.e., a link to the program detail page on edx.org).